Examples clean up

dbojicic opened the pull request:
Examples cleanup. 
Changes: 
- Organization, Practitioner, PractitionerRole changes to add HPI-O, HPI-Is, Medicare Provider Number respectively to align with HI Vendor Test Data
- Updated to use preferred terminology
- Added examples to demonstrate the use of MS elements
- QA fixes

dt-r commented on input/ig.xml:
```diff
@@ -398,6 +398,14 @@
 			<description value="Active prescription for Reaptan 10 mg/10 mg (perindopril arginine/amlodipine) tablet, 10."/>
 			<exampleCanonical value="http://hl7.org.au/fhir/core/StructureDefinition/au-core-medicationrequest"/>
 		</resource>
```
These comments are added on this file so that the conversation can be tracked and closed. They are not to change this file.
1. Missing cleanup on condition-ckd. This doesn't match AU Core, it's one of the american examples - if it is to be retained please adjust to be AU Core e.g. remove the additional meta values.

dbojicic replied:
1. Done
2. Removed & replaced with Ronny Irvine SA persona.
3. Removed example - it does not offer anything unique to the IG
4. Removed verificationStatus from two examples; inv ait-1 fails when both clinicalStatus&verificationStatus are absent
5. Done
6. Done
7. Done
8. Done - added to Procedure/stentinsertion

dt-r commented on input/ig.xml:
```diff
@@ -398,6 +398,14 @@
+			<name value="MedicationRequest - active prescription, Simvastatin"/>
+			<description value="Active prescription for Simvastatin 40 MG Disintegrating Oral Tablet."/>
+			<exampleCanonical value="http://hl7.org.au/fhir/core/StructureDefinition/au-core-medicationrequest"/>
+		</resource>
```
Missing additions of the new examples e.g.:
Observation/blood-group
Observation/blood-group-panel
Observation/rh-status

dbojicic replied:
Added

dt-r commented on input/examples/location-murrabit-hospital.xml:
```diff
@@ -1,3 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Compliant with HI Vendor Test Data -->
```
Remove the comment Compliant with HI Vendor Test Data for Location

dbojicic replied:
Done

dt-r commented on input/examples/location-mitchells-hill-audiology.xml:
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- Compliant with HI Vendor Test Data -->
```
Remove the comment Compliant with HI Vendor Test Data for Location. They do not store locations. We have made some choices on how to represent a location that is associated with data that is in the vendor test data environment however the statement that this example is compliant cannot be asserted.

dbojicic replied:
Done

dt-r commented on input/examples/practitioner-chau-fryer.xml:
```diff
@@ -1,9 +1,34 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- Compliant with HI Vendor Test Data -->
```
Any instance where 'Compliant with HI Vendor Test Data' will need to be explained. Not all elements are in the Vendor Test Data. If you want to keep this assertion in this tranche of work then a blurb will need to be crafted on which parts are compliant with. I would recommend that this goes into a GitHub backlog.

I've logged this once, but it applies to the other examples with this comment (except for Location that cannot assert compliance to HI Vendor Test Data)

dbojicic replied:
Removed from examples & added a GitHub backlog item.

dt-r commented on input/examples/practitioner-chau-fryer.xml:
```diff
@@ -1,9 +1,34 @@
+        <system value="http://ns.electronichealth.net.au/id/hi/hpii/1.0"/>
+        <value value="8003616566719020"/>
+    </identifier>
+    <identifier>
```
remove aphra. If you want to demonstrate some sort of other identifier, make a local one.

dbojicic replied:
Done

dt-r commented on input/examples/medicationrequest-reaptan.xml:
```diff
@@ -4,15 +4,24 @@
-            <display value="Reaptan 10 mg/10 mg (perindopril arginine/amlodipine) tablet, 10"/>
+          <system value="http://snomed.info/sct"/>
+          <code value="926213011000036100"/>
+          <display value="Reaptan 10 mg/10 mg (perindopril arginine/amlodipine) tablet, 10"/>
```
remove display either here or in one of the others. There's a lot of examples that are putting both display and text in CodeabelConcept elements.

dbojicic replied:
Done

dt-r commented on input/examples/practitioner-darleen-mackay.xml:
```diff
@@ -17,6 +19,18 @@
             <display value="Docker River Nephrology Clinic"/>
         </assigner>
```
Move the employee number to the PractitionerRole since it is role dependent.

dbojicic replied:
Done

dt-r reviewed (changes requested):
See inline comments. I've also created 2 backlog tasks in the project to improve the set of examples and improve the documentation.

mjosborne1 commented on input/examples/blood-group.xml:
recommend we change this to line up with laboratory practice, which is to report ABO and Rh group together in the one observation. Also, the responsible division of Pathology is not Chemistry, it's Blood Bank.
 <category>
        <coding>
            <system value="http://terminology.hl7.org/CodeSystem/v2-0074" />
            <code value="BLB" />
            <display value="Blood Bank" />
        </coding>
        <text value="Blood Bank" />
    </category>
    ...
        <code>
            <coding>
                <system value="http://loinc.org"/>
                <code value="882-1"/>
                <display value="ABO and Rh group [Type] in Blood"/>
            </coding>
            <text value="Blood Group"/>
        </code>
    <subject>
        <reference value="Patient/banks-mia-leanne" />
    </subject>
    <effectiveDateTime value="2024-05-10" />
            <valueCodeableConcept>
            <coding>
                <system value="http://snomed.info/sct"/>
                <code value="278149003"/>
                <display value="Blood group A Rh(D) positive"/>
            </coding>
            <text value="A Rh(D) Positive"/>
        </valueCodeableConcept>

dt-r replied:
@mjosborne1, you only need to comment. You wouldn't do a PR as you are a reviewer on _this_ PR.

dt-r replied:
That being said @mjosborne1, can link to the specific guidance stating to use Observation.component to represent this panel in FHIR over Observation.memberOf in FHIR.

mjosborne1 replied:
Perhaps I'm not understanding, but I don't think this is a use-case for components. What we have here are 2 tests  that are very often requested together. Group and Antibody Screen. There are 2 observations they are not components of a single observation. If this helps, the way I would represent this in V2 would be..
OBR|1||BGA^Blood Group and Antibody Screen|...
OBX|1|882-1^ABO and Rh group [Type] in Blood^LN||A Rh(D) Positive|
OBX|2|890-4^Blood group antibody screen [Presence] in Serum or Plasma^LN||Negative|
This may be better represented in Pathology Report resource.

dt-r replied:
@dbojicic - take these examples out of this PR. Add a specific project task that identifies what the goal is for this and work on it with @mjosborne1

mjosborne1 commented on input/examples/blood-group-panel.xml:
Similar to my comment on blood-group.xml, this is a good example of a commonly ordered panel, however the second result should be an antibody screen result, not rh-status. I'lll add an example in comments.
Also same comment about category / responsible division of pathology.
```
 <category>
        <coding>
            <system value="http://terminology.hl7.org/CodeSystem/v2-0074" />
            <code value="BLB" />
            <display value="Blood Bank" />
        </coding>
        <text value="Blood Bank" />
    </category>
```
Full result for antibody-screen.xml:
```
<?xml version="1.0" encoding="UTF-8"?>
<!-- Adjusted from FHIR Observation/bgpanel example-->
<Observation xmlns="http://hl7.org/fhir">
    <id value="antibody-screen"/>
    <meta>
        <profile value="http://hl7.org.au/fhir/core/StructureDefinition/au-core-diagnosticresult-path"/>
    </meta>
    <status value="final"/>
    <category>
        <coding>
            <system value="http://terminology.hl7.org/CodeSystem/observation-category" />
            <code value="laboratory" />
            <display value="Laboratory" />
        </coding>
    </category>
    <category>
        <coding>
            <system value="http://terminology.hl7.org/CodeSystem/v2-0074" />
            <code value="BLB" />
            <display value="Blood Bank" />
        </coding>
        <text value="Blood Bank" />
    </category>
        <code>
            <coding>
                <system value="http://loinc.org"/>
                <code value="890-4"/>
                <display value="Blood group antibody screen [Presence] in Serum or Plasma"/>
            </coding>
            <text value="Blood Group Antibody Screen"/>
        </code>
    <subject>
        <reference value="Patient/banks-mia-leanne" />
    </subject>
    <effectiveDateTime value="2024-05-10" />
            <valueCodeableConcept>
            <coding>
                <system value="http://snomed.info/sct"/>
                <code value="260385009"/>
                <display value="Negative"/>
            </coding>
            <text value="Negative"/>
        </valueCodeableConcept>
</Observation>
```

dt-r replied:
@dbojicic as above, separate this task from this PR.

dbojicic commented:
Addressed feedback from the first round of reviews, and made some additional changes to 
- remove examples that do not offer anything unique to the IG: Observation/bun-mcc, Location/mitchells-hill-audiology, Encounter/consultation
- Added Missing Data & Suppressed Data heading to the Examples page
- Removed Condition/cellulitis and added Condition/condition-masked example

dt-r commented on input/examples/condition-masked.xml:
```diff
@@ -1,5 +1,5 @@
 <Condition xmlns="http://hl7.org/fhir">
-  <id value="cellulitis"/>
+  <id value="condition-masked"/>
```
This is a poor example of masked. Only one element is masked. Redo.

dt-r replied:
change not in PR yet.

dbojicic replied:
Please check again

dt-r replied:
1. changes to condition-masked-subject: do not agree with adding a masked subject into AU Core example set. Suggest choose an Observation. If condition is kept, mask the category and code, and potentially other elements but not subject.
2. Addition of 2nd masked example not addressed - choose Observation, mask multiple fields including date.

dt-r replied:
1. Addressed
2. Addressed

dt-r commented on input/examples/condition-ckd.xml:
```diff
@@ -45,7 +42,7 @@
     </coding>
   </code>
```
Make the code 709044004 instead.

dt-r commented on input/examples/procedure-stentinsertion.xml:
```diff
@@ -32,6 +32,14 @@
         </coding>
         <text value="Occlusion of right coronary artery"/>
     </reasonCode>
+    <bodySite>
```
If this body site is added to this example you must update Procedure.code.text to include the body site information as per the guidance in AU core.

dbojicic replied:
Done

dt-r commented on input/ig.xml:
```diff
@@ -259,10 +259,10 @@
 			</reference>
-			<name value="Condition - Cellulitis of Right Knee"/>
-			<description value="Cellulitis of right knee&#xa;&#xa;Patient: Anne Bennelong"/>
+			<name value="Condition - Example with masked code"/>
```
Title and describe this as supressed data.

dt-r replied:
change not in PR yet.

dbojicic replied:
Please check again

dt-r commented on input/ig.xml:
```diff
@@ -563,10 +571,10 @@
-			<name value="Patient - Mr. Harry Dan"/>
-			<description value="Mr. Harry Dan"/>
+			<name value="Patient - Mr. Ronny Irvine"/>
+			<description value="Mr. Ronny Irvine"/>
```
Describe that this includes Missing Data

dt-r replied:
change not in PR yet.

dbojicic replied:
Please check again

dt-r replied:
Add (don't edit but add), two more examples for Missing Data, ensure one is a reference.

dbojicic replied:
> 1. On that example in which you have added Missing Data for MedicationRequest, also make the requester missing.
Added for the requester, please see
https://github.com/hl7au/au-fhir-core/blob/07b5867dbbbcc6115e431f6bbb613ff89204bb39/input/examples/medicationrequest-paracetamol-codeine.xml#L20

dt-r reviewed (changes requested):
See further inline comments.

JayMurdoch commented on input/examples/medication-2active-ingredients.xml:
```diff
@@ -5,6 +5,11 @@
+            <code value="13382H"/>
+            <display value="PERINDOPRIL + AMLODIPINE"/>
+        </coding>
+        <text value="perindopril arginine 10 mg + amlodipine 10 mg tablet, 30 Amlodipine"/>
```
Remove 'Amlodipine' after '... tablet, 30'

dbojicic replied:
Done

JayMurdoch commented on input/examples/medicationrequest-reaptan.xml:
```diff
@@ -4,15 +4,23 @@
+          <code value="926213011000036100"/>
         </coding>
-    </medicationCodeableConcept>
+        <text value="NReaptan 10 mg/10 mg (perindopril arginine/amlodipine) tablet, 10"/>
```
Remove 'N' from 'NReaptan'

dbojicic replied:
Done

JayMurdoch commented on input/examples/procedure-dialysis.xml:
```diff
@@ -18,7 +15,7 @@
     <text value="Hemodialysis"/>
```
Although SNOMED FSN is 'Hemodialysis (procedure), consider the use of AU preferred spelling/SNOMED preferred term for text value 'Haemodialysis'.

dbojicic replied:
Done

JayMurdoch replied:
No change in PR.

dbojicic replied:
Please check again

JayMurdoch reviewed (changes requested):
See inline comments. Thanks.

dt-r commented:
@dbojicic - see unresolved conversations for todos.

dt-r commented on input/ig.xml:
```diff
@@ -609,6 +609,14 @@
 			<description value="Deangelo Howe"/>
 			<exampleCanonical value="http://hl7.org.au/fhir/core/StructureDefinition/au-core-patient"/>
 		</resource>
+		<resource>
```
remove the term 'deidentified' - this is a use case not tackled, and you cannot declare no identifier and name to be 'deidentified'. Do not use a Patient resource as an example for suppressed data in AU Core until meaningful use case and design decisions have been made.

dt-r commented on input/examples/condition-ckd.xml:
```diff
@@ -3,9 +3,6 @@
   <!--  Example adapted from MCC eCare Plan Implementation Guide ConditionCKD  -->
```
Remove <!--  Example adapted from MCC eCare Plan Implementation Guide ConditionCKD  --> this now irrelevant, and also given the changes no longer true.

dbojicic replied:
Removed

dt-r commented on input/examples/procedure-dialysis.xml:
```diff
@@ -3,22 +3,19 @@
   <!--  Example adapted from MCC eCare Plan Implementation Guide Dialysis  -->
```
Remove  <!--  Example adapted from MCC eCare Plan Implementation Guide Dialysis  --> this is irrelevant.

dbojicic replied:
Done

dt-r commented on input/examples/medicationrequest-simvastatin.xml:
```diff
@@ -0,0 +1,53 @@
+    <status value="active"/>
+    <intent value="order"/>
+    <medicationReference>
+        <reference value="Medication/simvastatin"/>
```
Medication-simavastin is another non-AU core example that needs to be made AU Core. Clean it up, remove RX norm etc

dbojicic replied:
Done

dt-r reviewed (changes requested):
See new inline comments & unresolved conversations.

dt-r commented on input/pagecontent/examples.md:
```diff
@@ -15,7 +15,13 @@ In addition to the examples defined in this implementation, synthetic (realistic
+  * [MedicationRequest - prescription, paracetamol 500 mg + codeine phosphate hemihydrate 30 mg (missing data for status)](MedicationRequest-paracetamol-codeine.html) 
+* Suppressed Data:
+  * [Condition - Suppressed data: category and code](Condition-condition-masked.html)
+  * [Observation - Suppressed data: category, code, effectivePeriod and value](Observation-observation-masked.html)
```
State that the masked observation is a diagnostic result observation.

dt-r commented on input/pagecontent/examples.md:
```diff
@@ -15,7 +15,13 @@ In addition to the examples defined in this implementation, synthetic (realistic
-
+### Missing Data & Suppressed Data
+* Missing Data:
+  * [Patient - Ronny Irvine (missing data for birth date)](Patient-ronny-irvine.html)
```
The presentation of elements that are missing is different between Missing Data and Suppressed Data. Suggest that missing data presentation is revised.

dt-r reviewed (changes requested):
Previous comments resolved. See new inline comments.
